samples/solutions/producer_consumer.py

```diff
@@ -0,0 +1,244 @@
+import os
+import logging.handlers
+import glob
+
+# The interval at which to send telemetry
+TELEMETRY_INTERVAL = 10
+# Initial interval in seconds between consecutive connection attempts
+ATTEMPT_INTERVAL = 2
+# Threshold for retrying connection attempts after which the app will error
+THRESHOLD_FOR_RETRY_CONNECTION = 7200
+# Interval for rotating logs, in seconds
+LOG_ROTATION_INTERVAL = 3600
+# How many logs to keep before recycling
+LOG_BACKUP_COUNT = 6
+# Directory for storing log files
+LOG_DIRECTORY = "./logs"
+
+# Prepare the log directory
+os.makedirs(LOG_DIRECTORY, exist_ok=True)
+for filename in glob.glob("{}/*.log".format(LOG_DIRECTORY)):
+    os.remove(filename)
+
+log_formatter = logging.Formatter(
+    "%(asctime)s %(levelname)-5s (%(threadName)s) %(filename)s:%(funcName)s():%(message)s"
+)
+
+info_log_handler = logging.handlers.TimedRotatingFileHandler(
+    filename="{}/info.log".format(LOG_DIRECTORY),
+    when="S",
+    interval=LOG_ROTATION_INTERVAL,
+    backupCount=LOG_BACKUP_COUNT,
+)
+info_log_handler.setLevel(level=logging.INFO)
+info_log_handler.setFormatter(log_formatter)
+
+debug_log_handler = logging.handlers.TimedRotatingFileHandler(
+    filename="{}/debug.log".format(LOG_DIRECTORY),
+    when="S",
+    interval=LOG_ROTATION_INTERVAL,
+    backupCount=LOG_BACKUP_COUNT,
+)
+debug_log_handler.setLevel(level=logging.DEBUG)
+debug_log_handler.setFormatter(log_formatter)
+
+root_logger = logging.getLogger()
+root_logger.setLevel(level=logging.DEBUG)
+root_logger.addHandler(info_log_handler)
+root_logger.addHandler(debug_log_handler)
+
+sample_log_handler = logging.FileHandler(filename="{}/sample.log".format(LOG_DIRECTORY))
+sample_log_handler.setLevel(level=logging.DEBUG)
+sample_log_handler.setFormatter(log_formatter)
+logger = logging.getLogger(__name__)
+logger.addHandler(sample_log_handler)
+
+
+def get_type_name(e):
+    return type(e).__name__
+
+
+class Application(object):
+    async def initiate(self):
+        self.connected_event = asyncio.Event()
+        self.disconnected_event = asyncio.Event()
+        self.exit_app_event = asyncio.Event()
+        self.disconnected_event.set()
+
+        self.message_queue = asyncio.Queue()
+        self.device_client = None
+        self.first_connect = True
+        # Power factor for increasing interval between consecutive connection attempts.
+        # This will increase with iteration
+        self.retry_increase_factor = 1
+        # The nth number for attempting connection
+        self.sleep_time_between_conns = ATTEMPT_INTERVAL
+        self.try_number = 1
+
+    async def create_client(self, conn_str):
+        try:
+            # Create a Device Client
+            self.device_client = IoTHubDeviceClient.create_from_connection_string(conn_str)
+            # Attach the connection state handler
+            self.device_client.on_connection_state_change = self.handle_on_connection_state_change
+        except Exception as e:
```

Comment:
>except Exception as e: [](http://example.com/codeflow?start=8&length=22)

I'm surprised that you just swallow exceptions and continue here.

Reply:
yeah i should probably exit the app